tiledb/sm/array_schema/array_schema.cc

```diff
@@ -709,7 +738,7 @@ std::pair<uint64_t, uint64_t> ArraySchema::timestamp_range() const {
 
 URI ArraySchema::uri() {
   std::lock_guard<std::mutex> lock(mtx_);
-  if (uri_.is_invalid()) {
+  if (uri_.is_invalid() || new_uri_needed_) {
```

Comment:
Instead of using `new_uri_needed_`, can we update the URI in the `remove_attribute` function?

Reply:
yes, we can. I just wanted to do a lazy generation when we have multiple "remove_attribute". It should be fine if we generate uri immediately for remove_attribute or add_attribute